sockeye/arguments.py

```diff
@@ -148,11 +148,41 @@ def add_model_parameters(params):
                               type=int_greater_or_equal(0),
                               default=50000,
                               help='Maximum vocabulary size. Default: %(default)s.')
+    model_params.add_argument('--num-words-source',
+                              type=int_greater_or_equal(0),
+                              default=None,
+                              help='Maximum source vocabulary size. Overrides --num-words. Default: %(default)s')
+    model_params.add_argument('--num-words-target',
+                              type=int_greater_or_equal(0),
+                              default=None,
+                              help='Maximum target vocabulary size. Overrides --num-words. Default: %(default)s')
     model_params.add_argument('--word-min-count',
                               type=int_greater_or_equal(1),
                               default=1,
                               help='Minimum frequency of words to be included in vocabularies. Default: %(default)s.')
 
+    model_params.add_argument('--char-seq-encoder',
```

Comment:
Wrt. 2. If I understand it correctly, this not really a new encoder but a new "embedding" if you want. I.e. we could apply it to RNNs, transformers or other architectures.